astropy/io/fits/tests/test_hdulist.py

```diff
@@ -980,30 +972,26 @@ def test_no_resource_warning_raised_on_non_fits_file(self):
         #
         # Explicit tests are added to make sure the file handle is not
         # closed when passed in to fits.open. In this case the ResourceWarning
-        # was not raised, but a check is still included.
-        #
-        with catch_warnings(ResourceWarning) as ws:
-
-            # Make sure that files opened by the user are not closed
-            with open(filename, mode='rb') as f:
-                with pytest.raises(OSError):
-                    fits.open(f, ignore_missing_end=False)
+        # was not raised.
 
-                assert not f.closed
+        # Make sure that files opened by the user are not closed
+        with open(filename, mode='rb') as f:
+            with pytest.raises(OSError):
+                fits.open(f, ignore_missing_end=False)
 
-            with open(filename, mode='rb') as f:
-                with pytest.raises(OSError):
-                    fits.open(f, ignore_missing_end=True)
+            assert not f.closed
 
-                assert not f.closed
+        with open(filename, mode='rb') as f:
+            with pytest.raises(OSError), pytest.warns(VerifyWarning):
+                fits.open(f, ignore_missing_end=True)
 
-            with pytest.raises(OSError):
-                fits.open(filename, ignore_missing_end=False)
+            assert not f.closed
 
-            with pytest.raises(OSError):
-                fits.open(filename, ignore_missing_end=True)
+        with pytest.raises(OSError):
+            fits.open(filename, ignore_missing_end=False)
```

Comment:
Note: The diff is a bit misleading. All I did was un-indent after removing `with catch_warnings(ResourceWarning) as ws` because that catch is now unnecessary.